keel-clouddriver/src/main/kotlin/com/netflix/spinnaker/keel/clouddriver/MemoryCloudDriverCache.kt

```diff
@@ -68,29 +68,25 @@ class MemoryCloudDriverCache(
     }
 
   override fun securityGroupById(account: String, region: String, id: String): SecurityGroupSummary =
-    securityGroupSummariesById.getOrNotFound(
+    securityGroupSummariesByIdOrName.getOrNotFound(
       "$account:$region:$id",
       "Security group with id $id not found in the $account account and $region region"
     ) {
       val credential = credentialBy(account)
 
       // TODO-AJ should be able to swap this out for a call to `/search`
-      cloudDriver
-        .getSecurityGroupSummaries(account, credential.type, region, DEFAULT_SERVICE_ACCOUNT)
-        .firstOrNull { it.id == id }
+      cloudDriver.getSecurityGroupSummaryById(account, credential.type, region, id, DEFAULT_SERVICE_ACCOUNT)
     }
 
   override fun securityGroupByName(account: String, region: String, name: String): SecurityGroupSummary =
-    securityGroupSummariesById.getOrNotFound(
+    securityGroupSummariesByIdOrName.getOrNotFound(
       "$account:$region:$name",
       "Security group with name $name not found in the $account account and $region region"
     ) {
       val credential = credentialBy(account)
 
       // TODO-AJ should be able to swap this out for a call to `/search`
```

Comment:
Can probably lose this TODO comment now

Reply:
Right. I actually tried to look up this API but couldn't not connect the dots as `/search` doesn't seem to support returning security groups... Do you know anything more about it?

Reply:
Nope